### PR TITLE
Add missing examples for IpAddr enum

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @frewsxcv

### CODE REVIEW

File: src/libstd/net/ip.rs [hunk 1]
```diff
@@ -122,8 +167,23 @@ impl IpAddr {
...
     /// [IPv6]: ../../std/net/struct.Ipv6Addr.html#method.is_documentation
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(ip)]
+    ///
+    /// use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+    ///
+    /// fn main() {
```
Comment (frewsxcv) [inline src/libstd/net/ip.rs]: Why is there a `fn main() {` here?

Comment (frewsxcv) [inline src/libstd/net/ip.rs]: You should be able to omit the `fn main() {` even if you're using `#![feature(...)]`. An example:

https://github.com/rust-lang/rust/blob/b30022a1d32ebbd41f6623ab6fe7c62d489548e1/src/libcore/sync/atomic.rs#L205-L213

Comment (GuillaumeGomez) [inline src/libstd/net/ip.rs]: I know, I just prefer to keep it.

**Comment by frewsxcv** [comment]
📝 🎊 

@bors r+ rollup

**Pull request merged**